firebase-app-distribution/src/main/java/com/google/firebase/appdistribution/FirebaseAppDistribution.java

```diff
@@ -62,18 +125,119 @@ public static FirebaseAppDistribution getInstance() {
    * starts an installation If the latest release is an AAB, directs the tester to the Play app to
    * complete the download and installation.
    *
-   * @throws an {@link Status.UPDATE_NOT_AVAILABLE_ERROR} exception if no new release is cached from
-   *     checkForUpdate
+   * @throws FirebaseAppDistributionException with UPDATE_NOT_AVAIALBLE exception if no new release
+   *     is cached from checkForUpdate
    */
   @NonNull
   public UpdateTask updateApp() {
     return (UpdateTask) Tasks.forResult(new UpdateState(0, 0, UpdateStatus.PENDING));
   }
 
+  private boolean supportsCustomTabs(Context context) {
+    Intent customTabIntent = new Intent("android.support.customtabs.action.CustomTabsService");
+    customTabIntent.setPackage("com.android.chrome");
+    List<ResolveInfo> resolveInfos =
+        context.getPackageManager().queryIntentServices(customTabIntent, 0);
+    return resolveInfos != null && !resolveInfos.isEmpty();
+  }
+
+  private static String getApplicationName(Context context) {
+    try {
+      return context.getApplicationInfo().loadLabel(context.getPackageManager()).toString();
+    } catch (Exception e) {
+      Log.e(TAG, "Unable to retrieve App name");
+      return "";
+    }
+  }
+
+  private void openSignInFlowInBrowser(Uri uri) {
+    currentlySigningIn = true;
+    if (supportsCustomTabs(firebaseApp.getApplicationContext())) {
+      // If we can launch a chrome view, try that.
+      CustomTabsIntent customTabsIntent = new CustomTabsIntent.Builder().build();
+      Intent intent = customTabsIntent.intent;
+      intent.addFlags(Intent.FLAG_ACTIVITY_NO_HISTORY);
+      intent.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
+      customTabsIntent.launchUrl(currentActivity, uri);
+
+    } else {
+      // If we can't launch a chrome view try to launch anything that can handle a URL.
+      Intent browserIntent = new Intent(Intent.ACTION_VIEW, uri);
+      ResolveInfo info = currentActivity.getPackageManager().resolveActivity(browserIntent, 0);
+      browserIntent.addFlags(Intent.FLAG_ACTIVITY_NO_HISTORY);
+      browserIntent.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
+      currentActivity.startActivity(browserIntent);
+    }
+  }
+
+  private OnSuccessListener<String> getFidGenerationOnSuccessListener(Context context) {
+    return new OnSuccessListener<String>() {
+      @Override
+      public void onSuccess(String fid) {
+        Uri uri =
+            Uri.parse(
+                String.format(
```

Comment:
This should change to `https://appdistribution.firebase.dev/nba/pub/apps/"
                        + "%s/installations/%s/buildalerts?appName=%s&packageName=%s`